### PR TITLE
One client per actor

### PR DESCRIPTION
**Pull request by anschweitzer**
The key changes are in [actor_base.py](https://github.com/thegridelectric/gw-scada-spaceheat-python/blob/as/one-client-per-actor/gw_spaceheat/actors/actor_base.py), [scada_base.py](https://github.com/thegridelectric/gw-scada-spaceheat-python/blob/as/one-client-per-actor/gw_spaceheat/actors/scada_base.py) and [cloud_base.py](https://github.com/thegridelectric/gw-scada-spaceheat-python/blob/as/one-client-per-actor/gw_spaceheat/actors/cloud_base.py). The tests pass and a quick manual tests appeared to show that comm was restored after restarting mostquitto broker. 

Some defensive coding was added to tests:
1. Most have `try/finally` blocks so that the actors can be stopped even when an assertion fails the test. We (I) should make this more automatic in the future, but this is enough for the moment. 
2. I replaced some (most?) sleeps with calls to `wait_for()`. 
3. I commented out [test_atn_cli()](https://github.com/thegridelectric/gw-scada-spaceheat-python/blob/as/one-client-per-actor/test/test_examples.py#L153) because it was failing in a flaky way in CI and sometimes locally. 
4. Possibly related to 3, I wonder if [the BooleanActuator sensing loop](https://github.com/thegridelectric/gw-scada-spaceheat-python/blob/as/one-client-per-actor/gw_spaceheat/actors/boolean_actuator.py#L71) could use similar treatment to what we just did to SimpleSensor's loop? I'm not sure I'm following the code, if it looks fine to you I probably just am not absorbing it.

### CODE REVIEW

**Comment by anschweitzer** [comment]
`test_async_power_metering_dag()` remains flaky. It's unclear to me if is worse on in the PR.

File: gw_spaceheat/actors/actor_base.py [hunk 1]
```diff
@@ -71,33 +59,19 @@ def on_log(self, client, userdata, level, buf):
...
-    # noinspection PyUnusedLocal
-    def on_publish_disconnect(self, client, userdata, rc):
-        self.mqtt_log_hack([f"({helpers.log_time()}) Local Publish Disconnected! result code {str(rc)}"])
-    
-    # noinspection PyUnusedLocal
-    def on_consume_connect(self, client, userdata, flags, rc):
+    def on_connect(self, client, userdata, flags, rc):
         self.mqtt_log_hack(
-            [f"({helpers.log_time()}) Local Consume Connected flags {str(flags)} + result code {str(rc)}"]
+            [f"({helpers.log_time()}) Local Mqtt client Connected flags {str(flags)} + result code {str(rc)}"]
```
Comment (jessicamillar) [inline gw_spaceheat/actors/actor_base.py:64]: I got rid of references to "Consume" or "Publish" in the logs

File: .stickler.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 linters:
   flake8:
     python: 3
-    ignore: 'W28, W503'
+    ignore: 'W28, W503, E722'
```
Comment (jessicamillar) [inline .stickler.yml:4]: stickler is now globally ignoring bare `except`

**Comment by jessicamillar** [comment]
The `BooleanActuator` reporting is a little different than the `SimpleSensor`, because it reports on change as well as synchronously. Also, when the `reporting.SamplePeriodS` is equal to the `reporting.ReportingPeriodS` I want to offset the timing so that the scada isn't flushing its latest readings exactly at the same time it is getting a synchronous report. This issue will go away if we have a single actor. But I'll clean up the `BooleanActuator` and the pending `PowerMeter` so that what they are doing is easier to test and also more clear. In a different PR. (And also dig into the `test_async_power_metering_dag() ` flaikiness)

**Review by jessicamillar — COMMENTED** [review]
Looks good

**Pull request merged**